libsplinter/src/circuit/routing/mod.rs

```diff
@@ -234,8 +257,19 @@ impl CircuitNode {
     ///
     /// * `node_id` -  The unique ID for the circuit
     /// * `endpoints` -  A list of endpoints the node can be reached at
-    pub fn new(node_id: String, endpoints: Vec<String>) -> Self {
-        CircuitNode { node_id, endpoints }
+    pub fn new(
+        node_id: String,
+        endpoints: Vec<String>,
+        #[cfg(feature = "challenge-authorization")] public_key: Option<Vec<u8>>,
```

Comment:
Do we want public keys to be represented as Cylinder `PublicKey`s?

Reply:
The plan is to add a Splinter specific type to represent a Public key in these structs before challenge-authorization is stabilized. This struct will only have to/from bytes methods, any other operations would require converting to a Cylinder Public key.